src/BCReader.jl

```diff
@@ -211,21 +211,20 @@ function update_midmonth_data!(date, bcf_info::BCFileInfo{FT}) where {FT}
         bcf_info.monthly_fields[2] .= deepcopy(bcf_info.monthly_fields[1])
         bcf_info.segment_length .= Int(0)
 
-        # throw error when there are closer initial indices for the bc file data that matches this date0
-    elseif Dates.days(date - all_dates[Int(midmonth_idx + 1)]) > 2
+        # case 3: during initialization, warn and reset if there are closer file dates to the current date than the file date implied by midmonth_idx0+1
+    elseif (midmonth_idx == midmonth_idx0) && (date - all_dates[Int(midmonth_idx0) + 1]).value >= 0
```

Comment:
I'm a little confused about the logic here - It seems like `(date - all_dates[Int(midmonth_idx0) + 1]).value >= 0` implies that the current date is past the date of the initial index in the file, but does that mean that there's a closer file date to the current date than `all_dates[midmonth_idx0 + 1]`?

And the `midmonth_idx == midmonth_idx0` expression checks that we're on the initial date of the file, right? Maybe we could add some more comments clarifying these points

Reply:
Yes, case 3 is for when upon iniltialization, the simulation date is after the segment that was selected by the user (i.e. my date is 1 March, but the first segment I asked for was 1 Jan - 1 Feb)